lib/ansible/modules/net_tools/nsupdate.py

```diff
@@ -193,15 +209,15 @@ def __do_update(self, update):
             self.module.fail_json(msg='DNS server error: (%s): %s' % (e.__class__.__name__, to_native(e)))
         return response
 
-    def create_or_update_record(self):
+    def create_or_update_record(self, state, allow_duplicates):
         result = {'changed': False, 'failed': False}
 
         exists = self.record_exists()
         if exists in [0, 2]:
             if self.module.check_mode:
                 self.module.exit_json(changed=True)
 
-            if exists == 0:
+            if exists == 0 or allow_duplicates:
```

Comment:
@bcoca.. I don't know DNS well.  Is this all that's needed or do we have to check whether the record would be exactly the same as an existing record and not create it if so?

Reply:
depends ... some DNS servers allow multiple entry as a way to skew the load, i.e:
```
www.example.com    10.10.1.1
www.example.com    10.10.1.1
www.example.com    10.10.1.2
```
^ will send 2/3 of traffic to first 1 and 1/3 to the last .. so technically it should be accepted .. but that depends on the server software (some do this, some ignore, some error out).

Reply:
dupe are not standard, when you want to do load balancing with DNS you should use SRV records, or have a load balancer

Reply:
@nerzhul this 'feature' predates the RFC that introduced SRV and was used as a 'cheap' load balancing trick

Reply:
i really don't like a such crap in DNS nsupdate should take care about all DNS implementation , and bind9 is the reference and doesn't support a such thing

Reply:
The reason for me to want to add this is that I wanted to add multiple SRV records. There are also some other records that are, by default, supported to have multiple entries.
So one way is to allow users to add multiple of everything, which should be safe as long as they know what they're doing. But can be quite bad if someone who don't really know uses this feature.
But as we add this as an explicit flag which needs to be set, I think that by itself should trigger some thoughts in users.

I see three alternatives.
1. Be strict about this and only allow multiple records for the ones that allow it by default.
2. Same as (1) but also add more flags for special cases such as "DNS Load balancing"
3. Keep as is, and hope people know what they are doing.